plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureBlobContainerRetriesTests.java

```diff
@@ -391,6 +393,82 @@ public void testWriteLargeBlob() throws Exception {
         assertThat(blocks.isEmpty(), is(true));
     }
 
+    public void testWriteLargeBlobStreaming() throws Exception {
```

Comment:
These tests could be dried up more against the existing large blob write tests, but this is already quite the complicated PR since it touches all the plugins so I figured it best to clean up more in a follow-up and not touch too much of the existing code (I did dry things up in the production code a little more to make it easier to follow that this doesn't introduce much new interaction with the SDKs though).